tests/commands/test_commands.py

```diff
@@ -341,15 +342,74 @@ def json(self):
         return {'result': {'latest_block_height': self.height}}
 
 
+# @pytest.mark.execute
+# @patch('bigchaindb.lib.BigchainDB.get_validators')
+# @pytest.mark.abci
 @pytest.mark.skip
-@patch('bigchaindb.config_utils.autoconfigure')
-@patch('bigchaindb.backend.query.store_validator_update')
-@pytest.mark.tendermint
-def test_upsert_validator(mock_autoconfigure, mock_store_validator_update):
-    from bigchaindb.commands.bigchaindb import run_upsert_validator
+def test_upsert_validator_new_with_tendermint(b, priv_validator_path, user_sk, monkeypatch):
+    """WIP: Will be fixed and activated in the next PR
+    """
+    from bigchaindb.commands.bigchaindb import run_upsert_validator_new
+    import time
+
+    time.sleep(3)
+
+    def mock_get():
+        return [
+            {'pub_key': {'value': 'zL/DasvKulXZzhSNFwx4cLRXKkSM9GPK7Y0nZ4FEylM=',
+                         'type': 'tendermint/PubKeyEd25519'},
+             'voting_power': 10}
+        ]
+
+    # b.get_validators = mock_get
+    # mock_get_validators = mock_get
+    # monkeypatch.setattr('requests.get', mock_get)
+
+    proposer_key = b.get_validators()[0]['pub_key']['value']
+
+    args = Namespace(action='new',
+                     public_key=proposer_key,
+                     power=1,
+                     node_id='12345',
+                     sk=priv_validator_path,
+                     config={})
+    print("*****ARGS*****")
+    print(args)
+    resp = run_upsert_validator_new(args, b)
+    print("*****TX ID*****")
+    print(resp)
+    time.sleep(3)
+
+    assert b.get_transaction(resp)
 
-    args = Namespace(public_key='CJxdItf4lz2PwEf4SmYNAu/c/VpmX39JEgC5YpH7fxg=',
-                     power='10', config={})
-    run_upsert_validator(args)
 
-    assert mock_store_validator_update.called
+@pytest.mark.tendermint
+@pytest.mark.bdb
+def test_upsert_validator_new_without_tendermint(b, priv_validator_path, user_sk, monkeypatch):
+    from bigchaindb.commands.bigchaindb import run_upsert_validator_new
```

Comment:
Why is there an import statement inside this function definition?

Reply:
This is the code style used in all our other tests. I'm just trying to be consistent. I think it makes sense for tests, anyway. This way, if I run a single test, it depends on the minimal amount of imports. If I do all my imports at the top, then every test needs to import every module, whether it needs it or not.